### PR TITLE
Finished Frontend Tasks

### PR DESCRIPTION
**Pull request by ryan-truong**
**PREVIOUS** Main Changes
---
I have created an `API.js` file in the client folder and utilized axios to make API calls.

In the `CreateAccount.js` file, I used a simple password validation where the user can not click the create account button unless the passwords match. I then used the `post` route to upload the new users data, and confirmed this worked by running the server using `npm run dev` and going to `localhost:5000/users` and seeing that it was there.

In the `Login.js` file, I implemented an API call that uses the `post` route for `/login` on the server side. When implementing this, I kept getting `status 500` when trying to `post`, so I am not certain if the issue is on the client or server side.

Misc
---
Some minor changes to the pages were also done in this branch, specifcally:
* Adding a user profile upload option
* Adding a edit profile button on the user's page

---

**FINAL** Main Changes
---
Finished implementing all the front-end tasks needed for a functional website

On the events side this includes:
* Showing the list of events and having a page for each one
* Letting users create events
* Letting users attend events

On the user side this includes:
* Letting a user edit their information and upload a profile picture (also fixed it where only the user can edit their profile page)
* Showing all the events a user is attending

Misc:
* Navbar implemented where needed

### CODE REVIEW

File: client/src/API.js
Comment (shravanhariharan2) [inline client/src/API.js:24]: Should there be a route here in this call?

File: client/src/components/CreateAccount.js
Comment (shravanhariharan2) [inline client/src/components/CreateAccount.js:28]: I don't think we need to alert a user when their account has been created (I haven't really seen something like this in a standard login flow). I think just navigating should be fine

File: client/src/components/CreateAccount.js
Comment (shravanhariharan2) [inline client/src/components/CreateAccount.js:55]: Note that while it's fine to keep console.logs in this project since we aren't deploying it, we would normally remove log statements if we were to push this kind of application into production

**Review by shravanhariharan2 — APPROVED** [review]
Good stuff! Left some comments on code quality, let's wait for @farisashai's review to look more at the React-specific code before we merge. If he isn't able to get to reviewing it by the end of tomorrow then we can just merge this.

**Review by farisashai — APPROVED** [review]
nice work! this should help a lot with the demo. if you have time, we can implement a lot of small features

**Pull request merged**
